web/src/engine/providers/MangaPlugin.ts

```diff
@@ -76,6 +79,20 @@ export class DecoratableMangaScraper extends MangaScraper {
     public FetchImage(_page: Page, _priority: Priority, _signal: AbortSignal): Promise<Blob> {
         throw new Error();
     }
+
+    public async FilterChapters(list: Chapter[]): Promise<Chapter[]> {
+        return list.filter(
+            (obj, index) =>
+                list.findIndex((item) => item.Identifier === obj.Identifier) === index
+        );
+    }
+
+    public async FilterMangas(list: Manga[]): Promise<Manga[]> {
+        return list.filter(
+            (obj, index) =>
```

Comment:
One my want to simplfy this with generic types but i dont see how

```
    public async FilterChapters(list: Chapter[]): Promise<Chapter[]> {
        return this.FilterContent<Chapter>(list);

    }

    public async FilterMangas(list: Manga[]): Promise<Manga[]> {
        return this.FilterContent<Manga>(list);

    }

    private async FilterContent<T> (list : T[]) : Promise<T[]>{
        return list.filter(
            (obj, index) =>
                list.findIndex((item) => item.Identifier === obj.Identifier) === index
        );
    } 
```

Reply:
No need for generics, since manga and chapter are implementing the `IMediaContainer` interface which has the `Identifier` property.

Reply:
An alternative to `filter` is ustilizing a `set`
```typescript
const mangas = [{ id: '1' }, { id: '7' }, { id: '1' }, { id: '8' }, { id: '9' }];
const unique = [...new Set(mangas.map(manga => manga.id))].map(id => mangas.find(manga => manga.id === id));
```

Reply:
I wondered how to properly use Set for this <3